res/controllers/Korg-KAOSS-DJ-scripts.js

```diff
@@ -36,81 +61,85 @@ KAOSSDJ.deck = function(deckNumber) {
-            midi.sendShortMsg(ledChannels[i], led, ON);
-        }
-    }
+KAOSSDJ.sendAllKnobLeds = function(state) {
+    Object.values(MIDI_MAIN_KNOB_LEDS)
+        .forEach(led => midi.sendShortMsg(MIDI_CHANNELS.KNOB_MAIN, led, state));
+    Object.values([MIDI_CHANNELS.KNOB_LEFT, MIDI_CHANNELS.KNOB_RIGHT])
+        .forEach(channel => Object.values(MIDI_DECK_KNOB_LEDS).forEach(
+            led => midi.sendShortMsg(channel, led, state)));
+};
+
+KAOSSDJ.sendAllButtonLeds = function(state) {
+    Object.values([MIDI_CHANNELS.BUTTON_LEFT, MIDI_CHANNELS.BUTTON_RIGHT])
+        .forEach(channel => Object.values(MIDI_DECK_BUTTON_LEDS).forEach(
+            led => midi.sendShortMsg(channel, led, state)));
+
+};
+
+KAOSSDJ.init = function(_id, _debugging) {
+    // Turn on all knob LEDs and turn off all button LEDs.
+    KAOSSDJ.sendAllKnobLeds(MIDI_ON);
+    KAOSSDJ.sendAllButtonLeds(MIDI_OFF);
 
     // This message was copied from communication with Serato DJ Intro & Midi Monitor
     // It should setup mixxx from the controller defaults
-    var ControllerStatusSysex = [0xF0, 0x42, 0x40, 0x00, 0x01, 0x28, 0x00, 0x1F, 0x70, 0x01, 0xF7];
+    const ControllerStatusSysex = [0xF0, 0x42, 0x40, 0x00, 0x01, 0x28, 0x00, 0x1F, 0x70, 0x01, 0xF7];
     midi.sendSysexMsg(ControllerStatusSysex, ControllerStatusSysex.length);
 };
 
-KAOSSDJ.shutdown = function(id, debugging) {
-    // turn off all LEDs
-    for (var led = 0x00; led <= 0xFF; led++) {
-        for (key in ledChannel) {
-            midi.sendShortMsg(ledChannel[key], led, OFF);
-        }
-    }
+KAOSSDJ.shutdown = function(_id, _debugging) {
+    // Turn off all LEDs.
+    KAOSSDJ.sendAllKnobLeds(MIDI_OFF);
+    KAOSSDJ.sendAllButtonLeds(MIDI_OFF);
 };
 
 // ==== helper ====
 
 KAOSSDJ.getDeckIndexFromChannel = function(channel) {
-    return channel - 7
+    return channel - 7;
 };
 
 KAOSSDJ.getDeckByChannel = function(channel) {
-    var deckIndex = KAOSSDJ.getDeckIndexFromChannel(channel);
+    const deckIndex = KAOSSDJ.getDeckIndexFromChannel(channel);
     return KAOSSDJ.decks[deckIndex];
 };
 
 KAOSSDJ.updateDeckByChannel = function(channel, key, value) {
-    var deckIndex = KAOSSDJ.getDeckIndexFromChannel(channel);
+    const deckIndex = KAOSSDJ.getDeckIndexFromChannel(channel);
     KAOSSDJ.decks[deckIndex][key] = value;
 };
 
 // ==== mapped functions ====
 
-KAOSSDJ.wheelTouch = function(channel, control, value, status, group) {
-    var alpha = 1.0 / 8;
-    var beta = alpha / 32;
-    var deck = KAOSSDJ.getDeckByChannel(channel);
-    var deckNumber = deck.deckNumber;
-    var deck_playing = engine.getValue('[Channel' + deckNumber + ']', 'play_indicator');
+KAOSSDJ.wheelTouch = function(channel, _control, value, _status, _group) {
+    const alpha = 1.0 / 8;
+    const beta = alpha / 32;
+    const deck = KAOSSDJ.getDeckByChannel(channel);
+    const deckNumber = deck.deckNumber;
+    const deckPlaying = engine.getValue("[Channel" + deckNumber + "]", "play_latched");
```

Comment:
since you've rebased on 2.4, you can use template string literals.
```suggestion
    const deckPlaying = engine.getValue(`[Channel${deckNumber}]`, "play_latched");
```